libraries/lib-components/EffectInterface.h

```diff
@@ -114,37 +114,37 @@ class COMPONENTS_API EffectDefinitionInterface  /* not final */ : public Compone
    virtual ~EffectDefinitionInterface();
 
    //! Type determines how it behaves.
-   virtual EffectType GetType() = 0;
+   virtual EffectType GetType() const = 0;
```

Comment:
This PR fails to compile on macOS.  AudioUnitEffect is compiled only for that operating system, so a Windows build could succeed while the necessary changes to AudioUnitEffect were not done to make its methods const.

Even if you don't have a Mac available, it should not be difficult to amend each commit to build for Mac by using a test editor to fix AudioUnitEffect at each step.

Reply:
Alright, fixed now